src/extension.js

```diff
@@ -35,7 +35,17 @@ async function getPullRequests(context, showError) {
 		const showMerged = vscode.workspace.getConfiguration('pullRequestMonitor').get('showMerged');
 		const showClosed = vscode.workspace.getConfiguration('pullRequestMonitor').get('showClosed');
 		const count = vscode.workspace.getConfiguration('pullRequestMonitor').get('count');
-		const updatedPullRequests = await loadPullRequests(context.globalState.get('token'), { mode, showMerged, showClosed, repository, showError, count });
+		// configure the URL settings
+		let url = vscode.workspace.getConfiguration('pullRequestMonitor').get('githubEnterpriseUrl');
```

Comment:
Well I don't think that would be the right way to go about it, because most people don't know the github api endpoint for their enterprise server's in fact its not that well documented. Plus this is assuming people know what api your extension wants to use, which is not always the case, you're implementation would suggest having a solid understanding about how the underlining extension works, and communicates with github. Most users aren't contributors to the project, so they won't have a contributor level of knowledge going in.

I modeled it after how many other github api based extensions work such as: [vscode-github](https://github.com/KnisterPeter/vscode-github) as well as 
[vs-code-settings-sync](https://github.com/shanalikhan/code-settings-sync)

Reply:
Good point, I was concerned about user experience too. Then it could be extracted to a `getEndpointUrl(valueFromSettings)` function. This function would check if the setting value is null and return GitHub default url, otherwise append the path to the value from the setting.

The reason to extract it is because it needs to be implemented on the `loadRepositories` request as well.